### PR TITLE
"Spell-Jacking" mitigation ~ prevent sensitive data leak …

### PR DESCRIPTION
**Pull request by dlehammer**
… from spell checker.

Background: [Chrome & Edge Enhanced Spellcheck Features Expose PII, Even Your Passwords](https://www.otto-js.com/news/article/chrome-and-edge-enhanced-spellcheck-features-expose-pii-even-your-passwords)

### CODE REVIEW

**Comment by dlehammer** [comment]
I think perhaps username/password in [UserCreate.js](https://github.com/navidrome/navidrome/blob/aa21a2a305299483dc05f403e98dbc4b171fee0e/ui/src/user/UserCreate.js) and [UserEdit.js](https://github.com/navidrome/navidrome/blob/aa21a2a305299483dc05f403e98dbc4b171fee0e/ui/src/user/UserEdit.js) also should be adjusted, but I'm unsure of the syntax for `TextInput`.

Suggestions welcome :nerd_face:

**Comment by deluan** [comment]
I think you'll have to experiment with it, but AFAIK all attributes that TextInput does not interpret itself it passes down to the underlying Field/Input component.

**Comment by dlehammer** [comment]
Hi @deluan,

I'm unable to push to my PR branch, seems the cause is a linting check that fails for a unchanged file:
```
navidrome$ git push --force-with-lease 
go run github.com/golangci/golangci-lint/cmd/golangci-lint run -v --timeout 5m
INFO [config_reader] Config search paths: [./ /home/dleh/tmp/navidrome /home/dleh/tmp /home/dleh /home /] 
INFO [config_reader] Used config file .golangci.yml 
INFO [lintersdb] Active 25 linters: [asasalint asciicheck bidichk bodyclose depguard dogsled durationcheck errcheck errorlint exportloopref gocyclo goprintffuncname gosec gosimple govet ineffassign misspell nakedret nilerr rowserrcheck staticcheck typecheck unconvert unused whitespace] 
INFO [loader] Go packages loading at mode 575 (compiled_files|imports|name|types_sizes|deps|exports_file|files) took 1.858634245s 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 21.167995ms 
INFO [linters_context/goanalysis] analyzers took 34.91501088s with top 10 stages: gosec: 2.929829238s, bidichk: 2.244340557s, asasalint: 2.241053283s, buildir: 1.742389353s, commentmap: 1.590132777s, buildssa: 1.575711729s, directives: 1.57512492s, unused: 1.408026979s, whitespace: 1.281542912s, gocyclo: 1.121218228s 
WARN [linters_context] rowserrcheck is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649. 
INFO [runner] Issues before processing: 244, after processing: 1 
INFO [runner] Processors filtering stat (out/in): exclude-rules: 196/244, max_per_file_from_linter: 1/1, cgo: 244/244, filename_unadjuster: 244/244, autogenerated_exclude: 244/244, diff: 1/1, source_code: 1/1, severity-rules: 1/1, sort_results: 1/1, skip_files: 244/244, identifier_marker: 244/244, exclude: 244/244, path_shortener: 1/1, path_prefixer: 1/1, nolint: 185/196, uniq_by_line: 1/185, max_same_issues: 1/1, path_prettifier: 244/244, skip_dirs: 244/244, max_from_linter: 1/1 
INFO [runner] processing took 11.063996ms with stages: exclude-rules: 4.791827ms, identifier_marker: 2.596973ms, nolint: 1.802661ms, path_prettifier: 1.117686ms, autogenerated_exclude: 457.265µs, skip_dirs: 198.318µs, cgo: 36.196µs, filename_unadjuster: 28.112µs, source_code: 19.121µs, uniq_by_line: 11.089µs, max_same_issues: 1.483µs, max_from_linter: 708ns, path_shortener: 698ns, max_per_file_from_linter: 647ns, skip_files: 280ns, exclude: 241ns, diff: 212ns, sort_results: 196ns, severity-rules: 189ns, path_prefixer: 94ns 
INFO [runner] linters took 1.344659092s with stages: goanalysis_metalinter: 1.333495456s, rowserrcheck: 12.295µs 
scanner/metadata/taglib/get_filename.go:5:8: could not import C (cgo preprocessing failed) (typecheck)
import "C"
       ^
INFO File cache stats: 324 entries of total size 783.8KiB 
INFO Memory: 34 samples, avg is 136.7MB, max is 418.3MB 
INFO Execution took 3.231126716s                  
exit status 1
make: *** [Makefile:40: lint] Error 1
error: failed to push some refs to 'github.com:dlehammer/navidrome.git'
```

Any advice?

Thanks in advance :nerd_face:

**Comment by deluan** [comment]
Maybe you don't have the C build tools installed in your system? You can also skip verifications by using `git push --force-with-lease --no-verify`, and let the pipeline runs tests and linters for you.

**Comment by dlehammer** [comment]
Hi @deluan,

Based on the lint-check ["Code style issues found in 2 files. Forgot to run Prettier?"](https://github.com/navidrome/navidrome/actions/runs/4011156143/jobs/6888859107)
image

I've executed `npx prettier --write .` and rebased :nerd_face:

**Comment by dlehammer** [comment]
Regarding
> Maybe you don't have the C build tools installed in your system?

I followed the [Build from sources](https://www.navidrome.org/docs/installation/build-from-source/) in a clean virtual Ubuntu 22.04, didn't spot any C-related setup ~ only Go & Node.
I can execute `make buildall` successfully, albeit it seems the CI flow performs some additional checks via a Git hook.

**Comment by dlehammer** [comment]
Hi @deluan,

I've moved this to ready for review, as I've gotten as far as I can on my own.

I settled on the `spellCheck={false}` syntax, as the [spellCheck](https://17.reactjs.org/docs/dom-elements.html#all-supported-html-attributes) attribute is a 1st class React citizen.

My only issue is that React-admin (or React, what do I know :blush: ) populates the `spellcheck="false"` attribute on a higher-level div-element (pink marker) instead of the actual input element (blue marker), as illustrated in the create user "username" screenshot below.
I've also tried some other permutations, but regardless the propagation is on the div-element.
Screenshot from 2023-02-05 19-45-03


The [W3C - How TO - Disable Spellcheck for Forms](https://www.w3schools.com/howto/howto_html_spellcheck_disable.asp) examples adds the spellcheck attribute to the input field directly.
I've tested the behavior in Chrome and Firefox, and it seems the div-element attribute also deactivates spellcheck for the inner field.

I suggest this change is added as-is, as propagation is then delegated to the UI framework :nerd_face: 

(I noticed that Navidrome utilizes react-admin 3.18.3, and the latest available is 4.7.3 pt. Perhaps this issue is mitigated in v4+)

**Comment by deluan** [comment]
Thanks, for the PR. I'll merge it as is, I think it is enough to disable the spellcheck.

Re: ReactAdmin 4+, I'm planning a rewrite using it, as most of Navidrome's UI code is incompatible with it.

File: ui/src/user/UserCreate.js [hunk 1]
```diff
@@ -51,9 +51,13 @@ const UserCreate = (props) => {
...
-        <TextInput source="userName" validate={[required()]} />
+        <TextInput
+          spellCheck={false}
+          source="userName"
+          validate={[required()]}
+        />
         <TextInput source="name" validate={[required()]} />
-        <TextInput source="email" validate={[email()]} />
+        <TextInput spellCheck={false} source="email" validate={[email()]} />
         <PasswordInput source="password" validate={[required()]} />
```
Comment (deluan) [inline ui/src/user/UserCreate.js:61]: Don't we also need this in Password inputs?

Comment (dlehammer) [reply ui/src/user/UserCreate.js:61]: Spellcheck isn't enabled on `type="password"` inputs by default. This issue only affects password-inputs which use a "Show password" toggle to change the input to `type="text"`. I don't believe the Navidrome inputs do this, so there isn't a problem here AFAIK :nerd_face:

Comment (dlehammer) [reply ui/src/user/UserCreate.js]: Nice catch, I missed the little "eye" icon, toggling does indeed convert the password-input to a text-input.
I've amended create & edit user password fields (e0fabdab857d15c081d217f5857b9dc19cb7d471) :pray:

**Comment by dlehammer** [comment]
Hi @deluan,

Seems the CI [failed](https://github.com/dlehammer/navidrome/actions/runs/4102592439/jobs/7075743701) with an unrelated error:
```
Run ./update-translations.sh
Downloading Български (bg)
curl: (6) Could not resolve host: null
Error: Process completed with exit code 6.
```
Any ideas? :)

**Comment by deluan** [comment]
LGTM. Thanks!

**Pull request merged**